src/shogun/features/RandomFourierDotFeatures.cpp

```diff
@@ -90,15 +90,13 @@ SGVector<float64_t> RandomFourierDotFeatures::generate_random_parameter_vector()
 	SGVector<float64_t> vec(feats->get_dim_feature_space()+1);
 	switch (kernel)
 	{
-		case GAUSSIAN:
-			for (index_t i=0; i<vec.vlen-1; i++)
-			{
-				vec[i] = std::sqrt((float64_t)1 / kernel_params[0]) *
-				         std::sqrt(2.0) * normal_dist(m_prng);
-			}
-
+		case GAUSSIAN:{
+			auto parameter = std::sqrt((float64_t)1 / kernel_params[0]) *
+			std::sqrt(2.0) * normal_dist(m_prng);
```

Comment:
this is incorrect, you are now just generating a single random number, rather than drawing from the distribution `vlen` times.

Reply:
@NanuSai you could write a unit test before you refactor things, that helps catching such bugs

Reply:
True, but to generate random parameter for each instance we're forced to use loops (or is there a better way?) and if that's the case I think the current implementation is good.